src/blueprints/tv2/factories/tv2-viz-gfx-timeline-object-factory.ts

```diff
@@ -0,0 +1,114 @@
+import { Tv2GfxTimelineObjectFactory } from '../value-objects/factories/tv2-gfx-timeline-object-factory'
+import { Tv2BlueprintConfiguration } from '../value-objects/tv2-blueprint-configuration'
+import {
+  VizMseClearGfxTimelineObject,
+  VizMseContinueTimelineObject,
+  VizMseElementInternalTimelineObject,
+  VizMseLoadAllElementsTimelineObject,
+  VizType
+} from '../../timeline-state-resolver-types/viz-types'
+import { Tv2VizLayer } from '../value-objects/tv2-layers'
+import { DeviceType } from '../../../model/enums/device-type'
+import { TimelineObject } from '../../../model/entities/timeline-object'
+
+export class Tv2VizGfxTimelineObjectFactory implements Tv2GfxTimelineObjectFactory {
+  public createThemeOutTimelineObject(blueprintConfiguration: Tv2BlueprintConfiguration, duration: number): VizMseElementInternalTimelineObject {
+    return {
+      id: '',
+      enable: {
+        start: 0,
+        duration
+      },
+      priority: 100,
+      layer: Tv2VizLayer.GFX_ACTIONS,
+      content: {
+        deviceType: DeviceType.VIZMSE,
+        type: VizType.ELEMENT_INTERNAL,
+        templateName: 'OUT_TEMA_H',
+        templateData: [],
+        showName: blueprintConfiguration.showStyle.selectedGfxSetup.OvlShowName ?? ''
+      }
+    }
+  }
+
+  public createOverlayInitializeTimelineObject(duration: number): VizMseLoadAllElementsTimelineObject {
+    return {
+      id: 'loadAllElements',
+      enable: {
+        start: 0,
+        duration
+      },
+      priority: 100,
+      layer: Tv2VizLayer.GFX_ACTIONS,
+      content: {
+        deviceType: DeviceType.VIZMSE,
+        type: VizType.LOAD_ALL_ELEMENTS
+      }
+    }
+  }
+
+  public createContinueGfxTimelineObject(duration: number): VizMseContinueTimelineObject {
+    return {
+      id: '',
+      enable: {
+        start: 0,
+        duration
+      },
+      priority: 100,
+      layer: Tv2VizLayer.GFX_ACTIONS,
+      content: {
+        deviceType: DeviceType.VIZMSE,
+        type: VizType.CONTINUE,
+        direction: 1,
+        reference: Tv2VizLayer.GRAPHIC_PILOT     
+      }
+    }
+  }
+
+  public createGfxClearTimelineObject(blueprintConfiguration: Tv2BlueprintConfiguration, duration: number): VizMseClearGfxTimelineObject {
+    return (this.createBaseTimelineObjectWithContent(
+      duration,
+      this.createGfxClearContent(blueprintConfiguration)
+    ) as VizMseClearGfxTimelineObject) // Todo: figure out if this `as` usage is okay.
```

Comment:
Just ask us instead of comitting TODOs.

If you just follow the interface you defined in `Tv2GfxTimelineObjectFactory` then you don't need to cast at all.
All you need to do is change the return type from `VizMseClearGfxTimelineObject` to `TimelineOjbect` (which is what the factory interface returns) and you can get rid of the `as`.

Reply:
Extracted `content` type of the TimelineObject that was cast to, to its own interface, and used that as return type for `createGfxClearContent`, instead of `object`.

Also changed the return type as suggested, and removed the casting.